### PR TITLE
[cleanup] Remove AutoModelForVision2Seq + attention mask device

### PR DESCRIPTION
**Pull request by Wauplin**
- Removed `device` argument from `get_extended_attention_mask`. Device is inferred from `attention_mask`.
- `AutoModelForVision2Seq` is removed in favor of `AutoModelForImageTextToText`

Both were marked as deprecated and removed in v5. I've noticed them while working on some Hub cleanup. Feel free to ignore if not relevant or taken care of elsewhere.

---

Note: I've noticed a few other places with deprecated args. I'll take care of them in separate PR to avoid too many changes per PR.

### CODE REVIEW

**Comment by Cyrilvallez** [comment]
Superseded by https://github.com/huggingface/transformers/pull/43195!
